### PR TITLE
[DOCS] Model Zoo refactoring

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Changes

- Created a single models list at `./docs/ModelZoo.md` instead of list of models in the main `README.md` + list of models in every sample's `README.md`, resulting in duplication
- Refactored models tables to be in HTML format instead of Markdown, as it's more customizable

### Reason for changes

- Improving NNCF documentation
- Removing duplication in models tables
- Shortening main readme page

### CODE REVIEW

File: examples/tensorflow/object_detection/README.md
Comment (alexsu52) [inline examples/tensorflow/object_detection/README.md:163]: Please adjust the link in the "Run Object Detection Sample" paragraph. I would recommend adding direct links to pre-trained weights for RetinaNet and YOLO v4 in this README.
```
 Download the pre-trained weights in H5 format and provide the path to them using `--weights` flag. The link to the
archive with pre-trained weights can be found in the `TensorFlow checkpoint` column of the **[results](#results)** table.
```

File: examples/tensorflow/segmentation/README.md
Comment (alexsu52) [inline examples/tensorflow/segmentation/README.md:146]: Please adjust the link in the "Run Instance Segmentation Sample" paragraph. I would recommend adding direct links to pre-trained weights for Mask-R-CNN in this README.
```
- Download the pre-trained weights in checkpoint format and provide the path to them using `--weights` flag. The link to the
archive with pre-trained weights can be found in the `TensorFlow checkpoint` column of the [results](#results) table.
```

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: docs/ModelZoo.md
Comment (AlexanderDokuchaev) [inline docs/ModelZoo.md:20]: Fix indents (better use 2 spaces for html)

File: docs/ModelZoo.md
Comment (AlexanderDokuchaev) [inline docs/ModelZoo.md:36]: `style="text-align: center;"` is not needed. 
`<tbody align="center">` is enough.

File: docs/ModelZoo.md
Comment (AlexanderDokuchaev) [inline docs/ModelZoo.md:1]: `Model zoo` is it valide name for examples of using? may be better something like `Model Experiments` or 
`Model Results`?

File: examples/tensorflow/object_detection/README.md
Comment (nikita-savelyevv) [reply examples/tensorflow/object_detection/README.md:163]: Done

File: examples/tensorflow/segmentation/README.md
Comment (nikita-savelyevv) [reply examples/tensorflow/segmentation/README.md:146]: Done

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md]: Done

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md]: Done

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md:20]: Done

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md]: I agree that it looks cleaner, but results in a anchor symbol appearing when hovering over the header: 
image

I'd keep it as is because it looks the same and does not have this disadvantage

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md:1]: I guess it depends on the context. In the main README.md, this table was called **Model Zoo**, because it is common to name so such set of obtained models in other repositories. When these table were in each example's readme they could be referred to as **Results**, because they represented the results of the provided instructions. 

I would keep it as Model Zoo because we now mainly use it as in the first case.

**Review by kshpv — APPROVED** [review]

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md]: We decided to remove intermediate header for compression algorithms, and state "Quantization" explicitly

**Comment by AlexanderDokuchaev** [comment]
Example of align = left  for "model" and "Compression algorithm" collumns:
And `<li> </li>` tags for Compression algorithm (@nikita-savelyevv  how you think it's better than `<br>`?) 

<table>
  <thead>
    <tr>
      <th>Model</th>
      <th>Compression algorithm</th>
      <th>Dataset</th>
      <th>mAP&nbsp(<em>drop</em>)&nbsp%</th>
      <th>NNCF config file</th>
      <th>Checkpoint</th>
    </tr>
  </thead>
  <tbody align="center">
    <tr>
      <td align="left">RetinaNet</td>
      <td align="left"><li>Quantization: INT8 (per-tensor symmetric for weights, per-tensor asymmetric half-range for activations)</li><li>Filter pruning: 40%</li></td>
      <td>COCO&nbsp2017</td>
      <td>32.67 (0.76)</td>
      <td><a href="../examples/tensorflow/object_detection/configs/pruning_quantization/retinanet_coco_pruning_geometric_median_int8.json">Config</a></td>
      <td><a href="https://storage.openvinotoolkit.org/repositories/nncf/models/develop/tensorflow/retinanet_coco_pruning_geometric_median_int8.tar.gz">Download</a></td>
    </tr>
    <tr>
      <td align="left">RetinaNet</td>
      <td align="left">Quantization: INT8 (per-tensor symmetric for weights, per-tensor asymmetric half-range for activations)<br>Filter pruning: 40%</td>
      <td>COCO&nbsp2017</td>
      <td>32.67 (0.76)</td>
      <td><a href="../examples/tensorflow/object_detection/configs/pruning_quantization/retinanet_coco_pruning_geometric_median_int8.json">Config</a></td>
      <td><a href="https://storage.openvinotoolkit.org/repositories/nncf/models/develop/tensorflow/retinanet_coco_pruning_geometric_median_int8.tar.gz">Download</a></td>
    </tr>
    <tr>
      <td align="left">RetinaNet&#8209;Looooong</td>
      <td align="left"> - Quantization: INT8 (per-tensor symmetric for weights, per-tensor asymmetric half-range for activations)<br> - Filter pruning: 40%</td>
      <td>COCO&nbsp2017</td>
      <td>32.67 (0.76)</td>
      <td><a href="../examples/tensorflow/object_detection/configs/pruning_quantization/retinanet_coco_pruning_geometric_median_int8.json">Config</a></td>
      <td><a href="https://storage.openvinotoolkit.org/repositories/nncf/models/develop/tensorflow/retinanet_coco_pruning_geometric_median_int8.tar.gz">Download</a></td>
    </tr>
</table>

**Review by AlexanderDokuchaev — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]
Thanks for the contribution!

**Comment by alexsu52** [comment]
NNCF has the tool to update metrics by test results https://github.com/openvinotoolkit/nncf/blob/develop/tools/update_eval_results.py. After youer changes the tool should be updated. Could you file a ticket for this task?

**Comment by MaximProshin** [comment]
@ksilligan , why precommit tests were started for this PR if it only updates md files?

File: docs/ModelZoo.md
Comment (MaximProshin) [inline docs/ModelZoo.md:3]: I suggest to clearly state everywhere if PTQ or QAT was used (assume QAT in all cases now). Perhaps it's better to explain the abbreviations at the beginning and then use them in the tables, it will save some space.

File: docs/ModelZoo.md
Comment (MaximProshin) [inline docs/ModelZoo.md:4]: Let's add a note that PTQ accuracy and performance results for variety of networks can be found in https://docs.openvino.ai/latest/openvino_docs_performance_benchmarks.html

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md]: Done

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md:6]: Done

**Review by MaximProshin — APPROVED** [review]

**Review by vshampor — CHANGES_REQUESTED** [review]
I cannot approve it unless the auto-generation tool in tools/update_eval_results.py is updated and produces/updates the new layout in the HTML form. Otherwise we are losing the ability to easily keep the claimed results consistent with what we measure in nightlies.

File: docs/ModelZoo.md
Comment (nikita-savelyevv) [reply docs/ModelZoo.md]: Done

**Comment by alexsu52** [comment]
> I cannot approve it unless the auto-generation tool in tools/update_eval_results.py is updated and produces/updates the new layout in the HTML form. Otherwise we are losing the ability to easily keep the claimed results consistent with what we measure in nightlies.

I hope these changes can be done in the separate PR. It looks likes these changes is valuable for users.

**Comment by MaximProshin** [comment]
> > I cannot approve it unless the auto-generation tool in tools/update_eval_results.py is updated and produces/updates the new layout in the HTML form. Otherwise we are losing the ability to easily keep the claimed results consistent with what we measure in nightlies.
> 
> I hope these changes can be done in the separate PR. It looks likes these changes is valuable for users.

@vshampor , is this tool actual? Is there evidence that the tool works and is used? We discussed it with @alexsu52 and it seems we need to re-think the process with the updating of metrics and it seems to be taking additional effort but useless.

**Comment by MaximProshin** [comment]
> > > I cannot approve it unless the auto-generation tool in tools/update_eval_results.py is updated and produces/updates the new layout in the HTML form. Otherwise we are losing the ability to easily keep the claimed results consistent with what we measure in nightlies.
> > 
> > 
> > I hope these changes can be done in the separate PR. It looks likes these changes is valuable for users.
> 
> @vshampor , is this tool actual? Is there evidence that the tool works and is used? We discussed it with @alexsu52 and it seems we need to re-think the process with the updating of metrics and it seems to be taking additional effort but useless.

agreed to handle the issue with the tool in a separate ticket

**Pull request merged**
